cdap-app-fabric/src/main/java/co/cask/cdap/internal/provision/ProvisioningService.java

```diff
@@ -344,6 +344,8 @@ public Runnable provision(ProvisionRequest provisionRequest, DatasetContext data
                                     "plugin requirements. Following requirements were not meet by the listed " +
                                     "plugins: '%s'", programRunId.getProgram(), name,
                                   groupByRequirement(unfulfilledRequirements))));
+        programStateWriter.error(programRunId, new IllegalStateException("Provisioner does not meet all the " +
```

Comment:
nit: IllegalArgumentException fits a little better, since this is the user's fault.